conda/core/subdir_data.py

```diff
@@ -354,7 +354,17 @@ def _process_raw_repodata_str(self, raw_repodata_str):
         }
 
         channel_url = self.url_w_credentials
-        for fn, info in iteritems(json_obj.get('packages', {})):
+        legacy_packages = json_obj.get("packages", {})
+        conda_packages = json_obj.get("packages.conda", {})
+        _tar_bz2 = CONDA_PACKAGE_EXTENSION_V1
+        use_these_legacy_keys = set(iterkeys(legacy_packages)) - set(
+            k[:-6] + _tar_bz2 for k in iterkeys(conda_packages)
+        )
+
+        for fn, info in concatv(
+            iteritems(conda_packages),
+            ((k, legacy_packages[k]) for k in use_these_legacy_keys),
+        ):
```

Comment:
@msarahan if we want to add a feature flag to opt-in to using the new package format (which we can switch to `true` by default in a later conda release) then the only change we have to make to do that is localized right here.